dimod/binary_quadratic_model.py

```diff
@@ -2048,6 +2048,43 @@ def from_qubo(cls, Q, offset=0.0):
 
         return cls(linear, quadratic, offset, Vartype.BINARY)
 
+    @classmethod
+    def from_markov_random_field(cls, edges):
+        """
+
+
+        Examples:
+            >>> fixed = {'n01': 1, 'n03': 0. }
+            >>> edges = [['n01', 'n02', 3.0, 4.0, -6.0, 2.0], 
+                         ['n02', 'n04', 5.0, -1.0, -4.0, 2.0], 
+                         ['n03', 'n04', 1.0, 6.0, -6.0, 2.0], 
+                         ['n03', 'n05', -4.0, -3.0, 6.0, 5.0]]
+            ...
+            >>> # generate the markov network in a form that can be solved
+            >>> network = dimod.BinaryQuadraticModel.from_markov_random_field(edges)
+            ...
+            >>> # next we want to clamp/fix some of the variables
+            >>> clamped = network.copy()
+            >>> clamped.fix_variables(fixed)
+            ...
+            >>> # sample
+            >>> samples = dimod.ExactSolver().sample(clamped)
+            >>> samples.first
+            Sample(sample={'n02': 0, 'n04': 1, 'n05': 0}, energy=-1.0, num_occurrences=1)
```

Comment:
We'll have to include the additional step where the fixed variables are added back in so that the "full problem" is visible to the user on output. That might be best accomplished in a custom `to_dataframe` step.

Reply:
Yeah, I just added a simple `to_dataframe` method (see #334 ). For now it would be simple to append the df, but in the future we want to add a
```
samples = FixVariablesComposite(SimulatedAnnealingSampler()).sample(bqm)
```
which adds them back in to the record.

Reply:
You'd need to retain the original (non-fixed) `bqm` to do this, correct?  

Would this functionality be wrapped in a simple function? It's getting a bit convoluted for your average user off the street. :)

Reply:
The composite above would handle that for you (I edited it to make it a bit more clear). The idea it that the ising problem send to `SimulatedAnnealingSampler` would (in that example) have a/b clamped, but the returned `samples` would have those values reinserted

Reply:
I have expanded these thoughts in #335